Lollipop.md

stepan-lollipop opened the pull request:
#### Project Abstract

> Lollipop is a provider of modular solutions for Network Extensions/Solana L2/Rollup with diverse execution environments that can broadly be categorized as middlewares and instances. As opposed to Layer 2 solutions, Network Extensions does not result in liquidity or user fragmentation. Lollipop is an extension of Solana's monolithic infrastructure. NES environments can include TEEs, co-processors, zk-encryption, FHE, trusted solver networks, AI agents and more, even in non-blockchain contexts. Lollipop Instances and Middlewares do not rely on cross-chain technologies for integration with Solana. Instead, they interconnect via Solana itself, utilizing program delegation, preserving direct connectivity without bridges.
> Polkadot JAM in emulating SVM execution with PolkaVM, which is RISK-V level virtual-machine, so users may use SVM programs with Polkadot’s JAM, just like Mac users can run windows app with emulators like Parallels Desktop. With this being successful, the Execution Environment (VM) Fragmentation will be gone completely. sBPF Runtime as a Service in PolkaVM - is a middleware that will enable running SVM-related environments, taking full advantage of PVM and increasing Polkadot utilisation through synergies with the Solana and SVM-based ecosystems.
>

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`Lollipop.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (USDC & DOT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@qinwenwang:matrix.org` (change the homeserver if you use a different one)

stepan-lollipop commented:
I have read and hereby sign the Contributor License Agreement.

Noc2 reviewed (changes requested):
Thanks for the application. We will take a closer look as soon as possible. In the meantime, I noticed that the dot percentage is missing in your application, see https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#overview-1 Could you add this? We essentially need to pay teams in at least 50% vested DOT and let us know if you have any questions regarding this.

stepan-lollipop commented:
@Noc2 

Thanks for the complement!

> Thanks for the application. We will take a closer look as soon as possible. In the meantime, I noticed that the dot percentage is missing in your application, see https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#overview-1 Could you add this? We essentially need to pay teams in at least 50% vested DOT and let us know if you have any questions regarding this.

Added "DOT %: Percentage of Total Costs to be paid in vested DOT: 50%

stepan-lollipop commented:
@keeganquigley We would like to get it completely in the form of $DOT. So I've added additions clarifying that:
- **Total Costs:** $65,000 USD (equivalent in $DOT)
- **DOT %:** 100% in $DOT equivalent, Percentage of Total Costs to be paid in vested DOT: 50%

keeganquigley commented on applications/Lollipop.md:
```diff
@@ -0,0 +1,216 @@
+
+- **Team Name:** Lollipop Builders
+- **Payment Details:**
+  - **DOT**: 13W1E9hU8oKgVL1uaAGYTFLY9G1q2YJTjrVN5YEhUoXM4QHT, 
```
```suggestion
  - **DOT**: 13W1E9hU8oKgVL1uaAGYTFLY9G1q2YJTjrVN5YEhUoXM4QHT
```

keeganquigley reviewed (changes requested):
Thanks @stepan-lollipop sounds good. Could you also please specify a license in the milestone delivery table under 0a.? Currently all three are listed. Thanks!

Noc2 reviewed (approved):
Thanks for the update. I'm happy to go ahead with it.

keeganquigley reviewed (approved):
Thanks for the changes @stepan-lollipop happy to approve as well.

keeganquigley commented:
@stepan-lollipop while waiting for further reviews, could you please complete [KYB verification](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO) for Lollipop Builders? Thanks.

stepan-lollipop commented:
> @stepan-lollipop while waiting for further reviews, could you please complete [KYB verification](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO) for Lollipop Builders? Thanks.

Done, thank you!

stepan-lollipop commented:
> @stepan-lollipop while waiting for further reviews, could you please complete [KYB verification](https://in.sumsub.com/idensic/l/#/uni_JRqwo72Y5AjXaRDO) for Lollipop Builders? Thanks.

@keeganquigley done, thank you!

keeganquigley commented:
@stepan-lollipop please see my note in private chat regarding KYB.

semuelle reviewed (commented):
Hi @stepan-lollipop, thanks for the application. Happy to support it. The only issue is that we don't offer unvested DOT as payment option, so please only list total cost in USD and what percentage you would like paid in USDC vs vested DOT.

stepan-lollipop commented:
@semuelle Thanks for the clarification! Added the address for USDC in the application, added the USDC and DOT percentages, and made corrections in the Development Roadmap section on the appropriate lines.

PieWol commented:
I'm happy to support this. Sounds very interesting. 

Could you also integrate a Docker deliverable? It would be very convenient if the whole node boot and setup process would be covered with a Dockerfile. Probably makes it a lot easier to test and verify the outcomes of this grant. It even is one of our mandatory deliverables according to the application template. 

What do you think?

stepan-lollipop commented:
> I'm happy to support this. Sounds very interesting.
> 
> Could you also integrate a Docker deliverable? It would be very convenient if the whole node boot and setup process would be covered with a Dockerfile. Probably makes it a lot easier to test and verify the outcomes of this grant. It even is one of our mandatory deliverables according to the application template.
> 
> What do you think?

@PieWol  Yes, that is in our future plans. This will be a significant convenience for developers. So we are going to implement this in Docker later after after we finish prototyping and run sBPF with PVM implementation with JAM deployment.

PieWol commented:
@stepan-lollipop ,
It's great to hear that you plan on integrating it later on, but we want to have it as a deliverable for this grant already. Can you add a Docker deliverable that just allows to run "5. | Minimal sBPF Execution Test in PVM" in Docker? It's totally sufficient if it simply covers your progress that was made until the end of the grant. Thanks!

PieWol commented on applications/Lollipop.md:
```suggestion
| **0c.** | Testing and Testing Guide | Unit tests to measure compiled sBPF interpreter size |
| **0d.** | Docker | Containerize and run Minimal sBPF Execution Test in PVM using Docker |
| **0e.** | Article | Publish an article outlining feasibility analysis and prototype development |
| 1. | Compile `anza-xyz/sbpf` with 'no_std' and analyze feasibility | Analyze feasibility of `no_std` adaptation and remove Solana-specific dependencies |
| 2. | Memory & Execution Profiling for Minimal sBPF | Measure compiled binary size, execution time, and memory footprint for minimal sBPF programs. |
| 3. | Fork `anza-xyz/sbpf` & Introduce PVM Service (`jam-pvm-common`) | Refactor sBPF interpreter to work as a PolkaVM service using `jam-pvm-common`. |
| 4. | Compile & Measure sBPF Binary Size | Ensure it fits within 4MB PVM service limit. |
| 5. | Minimal sBPF Execution Test in PVM | Deploy and execute a simple contract (e.g., counter) in PolkaVM to confirm functional sBPF execution. The result of execution to be stored in a separated DB. |
```

takahser reviewed (changes requested):
@stepan-lollipop this looks very interesting.

However, could you clarify the following:

- **Account Model:** Are you planning to port Solana's account model as part of this grant, or is that a later-stage goal?
- **Parallel Tx Scheduling:** Solana processes transactions in parallel. Will your PolkaVM-based runtime also support concurrency, or is everything sequential at this stage?
- **Interoperability & Compatibility:** Could you talk a bit more about how far the compatibility goes? For example, will it be possible for a smartcontract on your VM to call a Solana smartcontract on their chain?

semuelle reviewed (approved):
Thanks for the update, happy to support.

takahser reviewed (approved):
@stepan-lollipop thx for the clarifications (chat) and updates. Happy to approve this.
